src/main/java/Duke.java

```diff
@@ -1,43 +1,96 @@
+                break;
+            }
+        }
+    }
+
+    private static void listAllTasks(Task[] tasks) {
+        for (int i = 0; i < nextAdd; i++) {
+//            StringBuilder sb = new StringBuilder();
+            String done = tasks[i].isPending() ? "[ ]" : "[X]";
+            System.out.println((i + 1) + ": " + tasks[i].getType() + done + " "
+                    + tasks[i].getTaskDesc() + " " + tasks[i].getAdditionalInfo());
+        }
+    }
+
+
     private static void userCommands() {
 
         boolean isProgramRunning = true;
         Scanner scanner = new Scanner(System.in);
         Task[] tasks = new Task[100];
-        int nextAdd = 0;
 
         while (isProgramRunning) {
             System.out.print("Enter your command here: ");
             String input = scanner.nextLine();
-            if (input.contains("done")) {
-                String[] words = input.split(" ");
-                for (int i = 1; i < words.length; i++) {
-                    int num = Integer.parseInt(words[i]) - 1;
-                    tasks[num].markAsDone();
-                    System.out.println("Congrats! You've completed this task: [X] "
-                            + tasks[num].getTaskDesc());
-                }
+            if (input.isEmpty()) {
+                //TODO: print something wrong
+            } else if (input.equals("help")) {
+                //TODO: print help statement
             } else if (input.equals("bye")) {
                 isProgramRunning = false;
             } else if (input.equals("list")) {
-                for (int i = 0; i < nextAdd; i++) {
-                    String done = tasks[i].isPending() ? "[ ]" : "[X]";
-                    System.out.println((i + 1) + ": " + done + " " + tasks[i].getTaskDesc());
-                }
+                listAllTasks(tasks);
             } else {
-                Task temp = new Task(input);
-                tasks[nextAdd] = temp;
-                System.out.println("Added: " + input);
-                nextAdd++;
+                String[] inputWords = input.split(" ");
+                if (inputWords[0].equals("done")) {
+                    makeTaskDone(inputWords, tasks);
+                } else {
+                    if (inputWords[0].equals("deadline")) {
+                        int num = Integer.parseInt(inputWords[1]) - 1;
+                        Deadline temp = new Deadline(tasks[num].getTaskDesc());
+                        String deadlineString = createRemainingString(inputWords);
+                        temp.setByDateTime(deadlineString);
+                        tasks[num] = temp;
+                        System.out.println("Deadline set: " + deadlineString);
+                    } else if (inputWords[0].equals("event")) {
+                        int num = Integer.parseInt(inputWords[1]) - 1;
+                        Event temp = new Event(tasks[num].getTaskDesc());
```

Comment:
Probably it is better to avoid the name temp since it cannot demonstrate the object type.